client/src/main/java/io/pravega/client/netty/impl/ClientConnectionInboundHandler.java

```diff
@@ -165,7 +176,38 @@ private Channel getChannel() throws ConnectionFailedException {
         }
         return ch;
     }
-    
+
+    /**
+     * This function completes the input future when the channel is registered.
+     *
+     * @param future CompletableFuture which will be completed once the channel is registered.
+     */
+    void completeWhenRegistered(final CompletableFuture<Void> future) {
```

Comment:
I have already created a class in common to solve this problem. 
Take a look at ReusableFutureLatch. It works like a reusable latch but instead of threads blocking on it, futures are registered with it, and when it is released all the features are completed. (Any late registrations are completed immediately)

Reply:
done.

Reply:
done, used reusableFutureLatch.